File: src/extras-kfilemetadata/libmimeutils.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KFileMetaData__MimeUtils
#include <QMimeDatabase>
#include <QMimeType>
#include <QString>
#include <mimeutils.h>
#include "libmimeutils.h"
#include "libmimeutils.hxx"

QMimeType* KFileMetaData__MimeUtils_StrictMimeType(const libqt_string filePath, const QMimeDatabase* db) {
    QString filePath_QString = QString::fromUtf8(filePath.data, filePath.len);
    return new QMimeType(KFileMetaData::MimeUtils::strictMimeType(filePath_QString, *db));
}

File: src/extras-kfilemetadata/libmimeutils.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const QMimeDatabase = @import("libqt6").QMimeDatabase;
const QMimeType = @import("libqt6").QMimeType;

/// ### [Upstream resources](https://api.kde.org/kfilemetadata-mimeutils.html)
pub const KFileMetaData__MimeUtils = extern struct {
    /// ### [Upstream resources](https://api.kde.org/kfilemetadata-mimeutils.html#strictMimeType)
    ///
    /// ## Parameter(s):
    ///
    /// ` filePath: []const u8 `
    ///
    /// ` db: QMimeDatabase `
    ///
    pub fn StrictMimeType(filePath: []const u8, db: anytype) QMimeType {
        const filePath_str = qtc.libqt_string{
            .len = filePath.len,
            .data = filePath.ptr,
        };
        comptime _ = @TypeOf(db)._is_QMimeDatabase;
        return .{ .ptr = qtc.KFileMetaData__MimeUtils_StrictMimeType(filePath_str, @ptrCast(db.ptr)) };
    }
};
